ssl/ech/ech_internal.c

```diff
@@ -1149,4 +1159,36 @@ int ossl_ech_intbuf_fetch(SSL_CONNECTION *s, unsigned char **buf, size_t *blen)
     *blen = s->ext.ech.transbuf_len;
     return 1;
 }
+
+int ossl_ech_stash_keyshares(SSL_CONNECTION *s)
+{
+    size_t i;
+
+    ech_free_stashed_key_shares(&s->ext.ech);
+    for (i = 0; i != s->s3.tmp.num_ks_pkey; i++) {
+        s->ext.ech.ks_pkey[i] = s->s3.tmp.ks_pkey[i];
+        EVP_PKEY_up_ref(s->ext.ech.ks_pkey[i]);
```

Comment:
Please check the up_ref result

Reply:
ah yes, you pointed that out before:-) added.